### PR TITLE
Content changes, validation, URLs

### PR DESCRIPTION
**Pull request by darbyo**

### CODE REVIEW

File: app/forms/UkAddressFormProvider.scala [hunk 1]
```diff
@@ -29,13 +29,21 @@ class UkAddressFormProvider @Inject() extends Mappings {
    def apply(): Form[UkAddress] = Form(
      mapping(
       "line1" -> text("ukAddress.error.line1.required")
-        .verifying(maxLength(250, "ukAddress.error.line1.length")),
+        .verifying(firstError(
+          maxLength(250, "ukAddress.error.line1.length"),
```
Comment (TheNickWilson) [inline app/forms/UkAddressFormProvider.scala:33]: Are we sure all of these fields should be 250 - that seems quite excessive?

**Review by TheNickWilson — CHANGES_REQUESTED** [review]

File: conf/messages.en [hunk 2]
```diff
@@ -57,44 +58,44 @@ unauthorised.heading = You can’t access this service with this account
...
+hasTradingName.checkYourAnswersLabel = Have a different trading name
+hasTradingName.error.required = Select yes if you use a different trading name in the UK
 hasTradingName.change.hidden = if your business trades using a name that is different from {0} in the UK
 
 partOfVatGroup.title = Is your business part of a UK VAT group?
 partOfVatGroup.heading = Is your business part of a UK VAT group?
 partOfVatGroup.guidance = VAT groups are when 2 or more companies or limited liability partnerships register as one ‘taxable’ person for VAT.
-partOfVatGroup.checkYourAnswersLabel = Is the business part of a UK VAT group?
-partOfVatGroup.error.required = Select yes if the business is part of a UK VAT group
+partOfVatGroup.checkYourAnswersLabel = Part of a VAT UK group
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:68]: Will update

File: conf/messages.en [hunk 3]
```diff
@@ -138,9 +139,9 @@ addEuDetails.remove.hidden = VAT details for {0}
 addEuDetails.addAnother = Add tax details for another EU country?
 addEuDetails.addAnother.hint = You must tell us your tax details for all EU countries.
 
-deleteEuVatDetails.title = Are you sure you want to remove the tax details for {0}?
-deleteEuVatDetails.heading = Are you sure you want to remove the tax details for {0}?
-deleteEuVatDetails.error.required = Select yes if you are sure you want to remove the tax details
+deleteEuVatDetails.title = Are you sure you need to remove the tax details for {0}?
+deleteEuVatDetails.heading = Are you sure you need to remove the tax details for {0}?
+deleteEuVatDetails.error.required = Select yes if you need to remove VAT details for this country
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:144]: Will double check with Mary

File: conf/messages.en [hunk 4]
```diff
@@ -401,118 +410,120 @@ alreadyRegistered.p1 = We will send you details about how to submit returns and
 alreadyRegistered.feedback.link.text = What did you think of this service?
 alreadyRegistered.feedback.30seconds = (takes 30 seconds)
 
-currentCountryOfRegistration.title = Which EU country is your business registered for the One Stop Shop in?
-currentCountryOfRegistration.heading = Which EU country is your business registered for the One Stop Shop in?
+currentCountryOfRegistration.title = Where is your business already registered for the One Stop Shop?
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:413]: Will confirm with Mary and update this

File: conf/messages.en [hunk 5]
```diff
@@ -401,118 +410,120 @@ alreadyRegistered.p1 = We will send you details about how to submit returns and
...
 bankDetails.bic = BIC
 bankDetails.bic.hint = Must be between 8 and 11 characters long. You can ask your bank or check your bank statement
-bankDetails.iban = IBAN number
-bankDetails.iban.hint = Can be up to 34 characters long. You can ask your bank or check your bank statement
+bankDetails.iban = IBAN
+bankDetails.iban.hint = Must be between 5 and 34 characters long. You can ask your bank or check your bank statement
 bankDetails.checkYourAnswersLabel = Bank account details
-bankDetails.error.accountName.required = Enter the account name
+bankDetails.error.accountName.required = Enter your account name
 bankDetails.error.bic.required = Enter the BIC number
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:471]: Will update to Your and confirm with Mary, we have decided against Swift code for now after Joel has done some investigation into that and thinks that we should have it removed

File: conf/messages.en [hunk 6]
```diff
@@ -401,118 +410,120 @@ alreadyRegistered.p1 = We will send you details about how to submit returns and
...
+currentlyRegisteredInCountry.checkYourAnswersLabel = Registered for One Stop Shop in EU country
 currentlyRegisteredInCountry.error.required = Select yes if your business is already registered for the One Stop Shop in {0}
 currentlyRegisteredInCountry.change.hidden = if your business is already registered for the One Stop Shop
 
 euTaxReference.title = What is your tax reference number for {0}?
 euTaxReference.heading = What is your tax reference number for {0}?
 euTaxReference.checkYourAnswersLabel = Tax reference number
-euTaxReference.error.required = Enter your tax reference number for {0}
-euTaxReference.error.length = EuTaxReference must be 100 characters or less
+euTaxReference.error.required = Enter your business tax reference number in {0}
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:491]: Will confirm with Mary and make sure they're consistent

File: conf/messages.en [hunk 7]
```diff
@@ -401,118 +410,120 @@ alreadyRegistered.p1 = We will send you details about how to submit returns and
...
-updateVatDetails.heading = You cannot use this service
-updateVatDetails.guidance = You will need to update the details of your VAT registration before registering for the One Stop Shop scheme.
+updateVatDetails.title = Update your VAT details
+updateVatDetails.heading = Update your VAT details
+updateVatDetails.guidance = Sign into your <a href="https://www.gov.uk/guidance/sign-in-to-your-hmrc-business-tax-account" class="govuk-link" rel="noreferrer noopener" target="_blank">business tax account (opens in a new tab)</a> to update your UK VAT details before registering for this scheme.
 
 businessAddressInUk.title = Is your principal place of business in the UK?
 businessAddressInUk.heading = Is your principal place of business in the UK?
-businessAddressInUk.checkYourAnswersLabel = Principal place of business in UK
+businessAddressInUk.checkYourAnswersLabel = Principal place of business address
```
[the remark this answers is not in the document]
Comment (darbyo) [reply conf/messages.en:526]: Havent seen this change, but again ill confirm with Mary

File: app/forms/UkAddressFormProvider.scala (hunk 1 above)
Comment (darbyo) [reply app/forms/UkAddressFormProvider.scala:33]: That is what the business confirmed as part of VEOSS-172, can check again if you think necessary?

File: conf/messages.en (hunk 7 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:526]: Apologies, changed this by mistake, have reverted back to previous.

File: conf/messages.en (hunk 2 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:68]: Have changed to Part of a UK VAT group

File: conf/messages.en (hunk 5 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:471]: Have changed to "Enter you BIC"

File: conf/messages.en (hunk 4 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:413]: Spoke with Mary, she said we just say it once at the start of their journey - confirm VAT details page. Can use short form from that point onwards.

File: conf/messages.en (hunk 3 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:144]: Changed this to "Select yes if you need to remove the tax details for {0}"

File: conf/messages.en (hunk 6 above)
[the remark this answers is not in the document]
Comment (warrenskiwork1979) [reply conf/messages.en:491]: Have put the question to Mary, she is awaiting a response from policy.

File: app/forms/UkAddressFormProvider.scala (hunk 1 above)
Comment (warrenskiwork1979) [reply app/forms/UkAddressFormProvider.scala:33]: Have put the question to Mary, awaiting a response.

**Pull request merged**